### PR TITLE
Create Provider to fetch settings everywhere

### PR DESCRIPTION
**Pull request by TonySma**

### CODE REVIEW

File: src/Provider/Settings.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusSettingsPlugin\Provider;
+
+use MonsieurBiz\SyliusSettingsPlugin\Exception\SettingsException;
+use MonsieurBiz\SyliusSettingsPlugin\Settings\RegistryInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+
+class Settings
```
Comment (maximehuran) [inline src/Provider/Settings.php:21]: You don't have interface for your Provider ?

Comment (TonySma) [reply src/Provider/Settings.php]: Yep sorry, too early in the morning ☕

**Review by maximehuran — CHANGES_REQUESTED** [review]

**Review by maximehuran — APPROVED** [review]

**Comment by lanfisis** [comment]
Thank you @TonySma for your help ! 🤟

File: README.md [hunk 2]
```diff
@@ -82,6 +82,25 @@ $ ./bin/console debug:container | grep app.settings.default
...
+    private SettingsProviderInterface $settingsProvider;
+
+    public function __construct(SettingsProviderInterface $settingsProvider)
+    {
+            $this->settingsProvider = $settingsProvider;
+    }
+    
+    public function myAwesomeMethod()
+    {
+        $this->settingsProvider->getSettingValue('app.default', 'demo_message')
```
Comment (maximehuran) [inline README.md:100]: Missing return

File: src/Twig/Extension/SettingsExtension.php [hunk 3]
```diff
@@ -54,12 +44,10 @@ public function getFunctions()
...
     {
         if (isset($context['sylius']) && $context['sylius'] instanceof ShopperContextInterface) {
-            if ($settingsInstance = $this->settingsRegistry->getByAlias($alias)) {
-                return $settingsInstance->getCurrentValue(
-                    $context['sylius']->getChannel(),
-                    $context['sylius']->getLocaleCode(),
-                    $path
-                );
+            try {
+                $this->settingsProvider->getSettingValue($alias, $path);
```
Comment (maximehuran) [inline src/Twig/Extension/SettingsExtension.php:48]: Missing return

**Pull request merged**